### PR TITLE
Add support for reading UEFI variables on windows

### PR DESCRIPTION
**Pull request by ChaitanyaKulkarni28**

### CODE REVIEW

File: google_guest_agent/uefi/uefi_windows.go [hunk 1]
```diff
@@ -0,0 +1,179 @@
...
+
+	// This call returns number of bytes written to the output buffer, unused, error
+	size, _, err := procGetFirmwareEnvironmentVariableW.Call(
+		uintptr(name),
+		uintptr(guid),
+		uintptr(unsafe.Pointer(&buffer[0])),
+		uintptr(uint32(len(buffer))),
+	)
+
+	if size == uintptr(0) {
```
Comment (dorileo) [inline google_guest_agent/uefi/uefi_windows.go:109]: Is there a condition of read size > 0 and  `error`?

File: google_guest_agent/uefi/uefi_windows.go [hunk 2]
```diff
@@ -0,0 +1,179 @@
...
+	}
+
+	return &Variable{
+		Name:       v,
+		Attributes: []byte{},
+		Content:    buffer[:size],
+	}, nil
+}
+
+// enablePrivilege enables the specified privilege for current process.
```
Comment (dorileo) [inline google_guest_agent/uefi/uefi_windows.go:120]: are we changing the guest agent's process? is it applicable to the runtime environment of the guest-agent "service"?

File: google_guest_agent/uefi/uefi_windows.go (hunk 1 above)
Comment (ChaitanyaKulkarni28) [reply google_guest_agent/uefi/uefi_windows.go:108]: no, thing with these calls is that `error` is always non-nil we need to inspect primary return value. Here return value 0 indicates the failure.

File: google_guest_agent/uefi/uefi_windows.go (hunk 2 above)
Comment (ChaitanyaKulkarni28) [reply google_guest_agent/uefi/uefi_windows.go:119]: yep, verified on `windows2019`

**Pull request merged**